### PR TITLE
Michael/ir 1983 registry - Post Processing and Highlight

### PR DESCRIPTION
**Pull request by MbfloydIR**
## Summary
Updated the post processing to use individual reactors that can be injected.
Added Highlight system to outline selected items using post processing.
Created unit tests for Post Processing Component

## References
closes #_insert number here_

### CODE REVIEW

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx [hunk 1]
```diff
@@ -23,29 +23,75 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
...
+import { CameraComponent } from '../../camera/components/CameraComponent'
+import { HighlightState } from '../HighlightState'
+import { RendererState } from '../RendererState'
+import { RendererComponent } from '../WebGLRendererSystem'
+import { ObjectLayers } from '../constants/ObjectLayers'
+import { PostProcessingEffectState } from '../effects/EffectRegistry'
+import { LinearTosRGBEffect } from '../effects/LinearTosRGBEffect'
 import { useScene } from './SceneComponents'
 
+declare module 'postprocessing' {
```
Comment (HexaField) [inline packages/spatial/src/renderer/components/PostProcessingComponent.tsx:47]: this effect type override should be moved to each of the respective modules

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx [hunk 2]
```diff
@@ -23,29 +23,75 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
...
+
 export const PostProcessingComponent = defineComponent({
   name: 'PostProcessingComponent',
   jsonID: 'EE_postprocessing',
 
   onInit(entity) {
     return {
       enabled: false,
-      effects: defaultPostProcessingSchema
+      effects: {} as Record<string, any> // effect name, parameters
```
Comment (HexaField) [inline packages/spatial/src/renderer/components/PostProcessingComponent.tsx]: can we add a better type than `any` for the record value here? we could a base type without effect-specific props perhaps for at least the context of this module

File: packages/spatial/src/renderer/effects/EffectRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import { Entity } from '@etherealengine/ecs'
+import { defineState } from '@etherealengine/hyperflux'
+import { EffectComposer } from 'postprocessing'
+import React from 'react'
+import { Scene } from 'three'
+
+export type EffectReactorProps = {
+  isActive: boolean
+  rendererEntity: Entity
+  effectData
```
Comment (HexaField) [inline packages/spatial/src/renderer/effects/EffectRegistry.ts]: let's provide types for `effectData` and `effects` here, even if they are just `any`

File: packages/spatial/tsconfig.json [hunk 4]
```diff
@@ -41,5 +41,5 @@
     "../ecs/**/*",
     "./**/*.ts",
     "./**/*.tsx"
-  ]
```
Comment (HexaField) [inline packages/spatial/tsconfig.json]: please check this is correct as per our indentation

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx (hunk 1 above)
Comment (MbfloydIR) [reply packages/spatial/src/renderer/components/PostProcessingComponent.tsx:47]: implemented

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx (hunk 2 above)
Comment (MbfloydIR) [reply packages/spatial/src/renderer/components/PostProcessingComponent.tsx]: implemented

File: packages/spatial/src/renderer/effects/EffectRegistry.ts (hunk 3 above)
Comment (MbfloydIR) [reply packages/spatial/src/renderer/effects/EffectRegistry.ts]: implemented

File: packages/spatial/tsconfig.json (hunk 4 above)
Comment (MbfloydIR) [reply packages/spatial/tsconfig.json]: implemented

**Review by HexaField — APPROVED** [review]

**Pull request merged**